ha/core/controllers/pcs/node_controller.py

```diff
@@ -50,16 +50,65 @@ def initialize(self, controllers):
         raise HAUnimplemented("This operation is not implemented.")
 
     @controller_error_handler
-    def start(self, nodeid: str) -> dict:
+    def start(self, node_id: str, **op_kwargs) -> dict:
         """
-        Start node with nodeid.
+        Start node with the node_id.
         Args:
-            nodeid (str): Node ID from cluster nodes.
+            node_id (str): Node ID from cluster nodes.
         Returns:
             ([dict]): Return dictionary. {"status": "", "output": "", "error": ""}
                 status: Succeeded, Failed, InProgress
         """
-        raise HAUnimplemented("This operation is not implemented.")
+        try:
+            # Get the node_name (pvtfqdn) from node_id
+            node_name = self._get_node_name(node_id=node_id)
+            self._is_node_in_cluster(node_id=node_name)
+            # TODO: Check status from system health instead of pcs below.
+            node_status = self.nodes_status([node_name])[node_name]
+            Log.debug(f"Node {node_name} status is {node_status}")
+            if node_status == NODE_STATUSES.ONLINE.value:
+                Log.debug(f"Node {node_name} is already online")
+                return {"status": const.STATUSES.SUCCEEDED.value, "output": NODE_STATUSES.ONLINE.value, "error": ""}
+            elif node_status == NODE_STATUSES.STANDBY.value or node_status == NODE_STATUSES.STANDBY_WITH_RESOURCES_RUNNING.value:
+                # Unstandby the node
+                if self.heal_resource(node_name):
+                    _output, _err, _rc = self._execute.run_cmd(const.PCS_NODE_UNSTANDBY.replace("<node>", node_name), check_error=False)
+                    if _rc != 0:
+                        Log.error(f"Failed to start node {node_name}, Error: {_err}")
+                        return {"status": const.STATUSES.FAILED.value, "output": "", "error": f"Failed to start node {node_id}, Error: {_err}"}
+                    Log.debug(f"Node {node_name} was in standby mode, unstandby operation started successfully")
+                else:
+                    Log.error(f"Node {node_name} is in standby mode : Resource failcount found on the node, cleanup did not work")
+                    return {"status": const.STATUSES.FAILED.value, "output": "", "error": f"Node {node_id} is in standby mode, resource failcount found on the node, cleanup did not work"}
+            elif node_status == NODE_STATUSES.CLUSTER_OFFLINE.value:
+                _output, _err, _rc = self._execute.run_cmd(const.PCS_NODE_START.replace("<node>", node_name), check_error=False)
+                if _rc != 0:
+                    Log.error(f"Failed to start node {node_name}, Error: {_err}")
+                    return {"status": const.STATUSES.FAILED.value, "output": "", "error": f"Failed to start node {node_id}, Error: {_err}"}
+                Log.debug(f"Node {node_name} started successfully. Waiting for cluster to stabalize and then get the node status")
```

Comment:
Do we also need to verify that resources have started on this node ?

Reply:
I think no, we just start the pcs cluster on that node.

Reply:
The jira description mentions that :
>>2. All cluster managed cortx services should be started after node is powered on.
so asking

Reply:
Yes, right. We start the pcs cluster which eventually will start the services.